paasta_tools/deployd/queue.py

```diff
@@ -107,21 +148,43 @@ def _lock_and_get_entry(self, entry_node):
             self.client.delete(lock_path)
             return None
 
-    def _consume(self, entry_node):
+    def _consume(self, entry_node: str):
         # necessary in case we lose connection at some point
         if not self._holds_lock(entry_node):
             return  # TODO: log?
         with self.client.transaction() as transaction:
             transaction.delete(f"{self.locks_path}/{entry_node}")
             transaction.delete(f"{self.entries_path}/{entry_node}")
 
-    def _holds_lock(self, entry_node):
+    def _holds_lock(self, entry_node: str):
         lock_path = f"{self.locks_path}/{entry_node}"
         self.client.sync(lock_path)
         value, stat = self.client.retry(self.client.get, lock_path)
         return value == self.id
 
-    def _release(self, entry_node):
+    def _release(self, entry_node: str):
         if not self._holds_lock(entry_node):
             return
         self.client.delete(f"{self.locks_path}/{entry_node}")
+
+    def get_available_service_instances(
+        self
+    ) -> Iterable[Tuple[float, ServiceInstance]]:
+        self._update_local_state(None)
```

Comment:
Do you actually need to do this?  Since we have a watch, I think that these should just stay up to date, right?  You might just want to put this method in the condition context manager so that stuff doesn't change while you're looking at it.

Reply:
Yeah, I should probably make a copy inside the context manager, good call.

Like for `get`, I added `_update_local_state` for the benefit of the tests; it's nice to be able to `put()` and then immediately `get_available_service_instances` and have that include what you put. Maybe I'll just make the tests explicitly call `_update_local_state`